src/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -906,33 +917,38 @@ public static string Format(Type enumType, object value, string format)
                 }
                 return ((Enum)value).ToString(format);
             }
+            
             // The value must be of the same type as the Underlying type of the Enum
-            else if (valueType != underlyingType)
+            Type underlyingType = GetUnderlyingType(enumType);
+            if (valueType != underlyingType)
             {
                 throw new ArgumentException(SR.Format(SR.Arg_EnumFormatUnderlyingTypeAndObjectMustBeSameType, valueType.ToString(), underlyingType.ToString()));
             }
-            if (format.Length != 1)
-            {
-                // all acceptable format string are of length 1
-                throw new FormatException(SR.Format_InvalidEnumFormatSpecification);
-            }
 
-            char formatCh = format[0];
-            if (formatCh == 'G' || formatCh == 'g')
-                return GetEnumName(rtType, ToUInt64(value)) ?? value.ToString();
+            if (format.Length == 1)
```

Comment:
```csharp
if (format.Length != 1)
 throw new FormatException(SR.Format_InvalidEnumFormatSpecification);

switch (format[0])
...
```

?

Reply:
We also need to throw the same exception if the format[0] is an unknown char.

Reply:
(Same in one more place.)

Reply:
I see - then ignore my comment.